### PR TITLE
bumps parser version

### PR DESCRIPTION
**Pull request by brunocalza**
# Summary

Bumps parser version to `de3c827`

# Context

It includes features such as all [`JOIN` operators](https://github.com/tablelandnetwork/go-sqlparser/pull/33), [compound queries](https://github.com/tablelandnetwork/go-sqlparser/pull/34), and [`INSERT` with `SELECT`](https://github.com/tablelandnetwork/go-sqlparser/pull/36).

# Implementation overview

- There was a refactoring in how the query validation is done to use the helpers from the parser package. The reason for that is so that the validator and the SDK can both rely on the same validation
- It was added the extra validation for an `INSERT` with `SELECT` query that checks if the chain of the insert table is the same chain of the select table, finishing the feature

### CODE REVIEW

File: pkg/parsing/impl/validator.go [hunk 1]
```diff
@@ -62,33 +61,24 @@ func (pp *QueryValidator) ValidateCreateTable(query string, chainID tableland.Ch
 	}
 
 	node := stmt.(*sqlparser.CreateTable)
-	if !pp.createTableNameRegEx.MatchString(node.Table.String()) {
-		return nil, &parsing.ErrInvalidTableName{}
-	}
-	parts := strings.Split(node.Table.String(), "_")
-	if len(parts) < 2 {
-		return nil, fmt.Errorf("table name isn't referencing the chain id")
+	validTable, err := sqlparser.ValidateCreateTargetTable(node.Table)
```
Comment (brunocalza) [inline pkg/parsing/impl/validator.go:64]: Using the parser package method to validate the table of a CREATE statement

File: pkg/parsing/impl/validator.go [hunk 2]
```diff
@@ -398,24 +353,52 @@ func (s *readStmt) GetQuery() (string, error) {
...
+		return nil, fmt.Errorf("no system-table reference: %w", err)
+	}
+
+	insertTable, err := sqlparser.ValidateTargetTable(stmt.GetTable())
+	if err != nil {
+		return nil, fmt.Errorf("table name is not valid: %w", err)
 	}
 
-	return stmt.GetTable().String(), nil
+	if insert, ok := stmt.(*sqlparser.Insert); ok && insert.Select != nil {
```
Comment (brunocalza) [inline pkg/parsing/impl/validator.go:366]: If it is an INSERT with SELECT do the chain check

File: pkg/parsing/impl/validator_test.go [hunk 3]
```diff
@@ -265,12 +265,14 @@ func TestWriteQuery(t *testing.T) {
 			expErrType: ptr2ErrInvalidSyntax(),
 		},
 
-		// Disallow JOINs and sub-queries
+		// insert with select chain mismatch
 		{
-			name:       "insert subquery",
```
Comment (brunocalza) [inline pkg/parsing/impl/validator_test.go:270]: This is the test to check for the chain mismatch error

File: pkg/parsing/impl/validator_test.go [hunk 4]
```diff
@@ -166,7 +166,7 @@ func TestWriteQuery(t *testing.T) {
 		{
 			name:       "table id or chain id is missing",
 			query:      "delete from Hello_4 where a=2",
-			expErrType: ptr2ErrInvalidTableName(),
+			expErrType: ptr2ErrWrongFormatTableName(),
```
Comment (brunocalza) [inline pkg/parsing/impl/validator_test.go:169]: gotta change the error to use the error returned by the parser package

File: pkg/parsing/impl/validator.go (hunk 1 above)
Comment (brunocalza) [reply pkg/parsing/impl/validator.go:64]: Most of the changes in this file is do adjust the code to the fact that the validation methods return the types `ValidatedTable` and `ValidatedCreateTable`

File: pkg/eventprocessor/impl/eventprocessor_replayhistory_test.go [hunk 5]
```diff
@@ -36,12 +36,12 @@ func TestReplayProductionHistory(t *testing.T) {
 
 	expectedStateHashes := map[tableland.ChainID]string{
 		1:      "ce4f083e256d3458a329b6cd1ba7d8e93d9703b3",
-		5:      "9ab5a92ac961d39ee07dd1e59518c527b39719c1",
-		10:     "3d907df6936eea9ab046d89c80b995d5db26e6ac",
+		5:      "b42740b92e40d616061a88872e19616c05469750",
```
Comment (brunocalza) [inline pkg/eventprocessor/impl/eventprocessor_replayhistory_test.go:39]: i'm not really sure why a change in the state happened. maybe an `INSERT` with `SELECT` query was accepted?

**Review by jsign — APPROVED** [review]
LGTM.

Left some comments reg the e2e test. I suggest some idea to try to get more confidence if there's something hairy going on. If we aren't sure what's happening, that's a bit concerning since we might discover in production.

**Pull request merged**